cpp/src/plasma/lib/java/org_apache_arrow_plasma_PlasmaClientJNI.cc

```diff
@@ -0,0 +1,314 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "plasma/lib/java/org_apache_arrow_plasma_PlasmaClientJNI.h"
+
+#include <pthread.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <algorithm>
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "plasma/client.h"
+
+const jsize LEN_OF_OBJECTID = sizeof(plasma::ObjectID) / sizeof(jbyte);
+
+inline void jbyteArray_to_object_id(JNIEnv* env, jbyteArray a, plasma::ObjectID* oid) {
+  env->GetByteArrayRegion(a, 0, LEN_OF_OBJECTID, reinterpret_cast<jbyte*>(oid));
+}
+
+inline void object_id_to_jbyteArray(JNIEnv* env, jbyteArray a, plasma::ObjectID* oid) {
+  env->SetByteArrayRegion(a, 0, LEN_OF_OBJECTID, reinterpret_cast<jbyte*>(oid));
+}
+
+class JByteArrayGetter {
+ private:
+  JNIEnv* _env;
+  jbyteArray _a;
+  jbyte* bp;
+
+ public:
+  JByteArrayGetter(JNIEnv* env, jbyteArray a, jbyte** out) {
+    _env = env;
+    _a = a;
+
+    bp = _env->GetByteArrayElements(_a, NULL);
+    *out = bp;
+  }
+
+  ~JByteArrayGetter() { _env->ReleaseByteArrayElements(_a, bp, 0); }
+};
+
+JNIEXPORT jlong JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_connect(
+    JNIEnv* env, jclass cls, jstring store_socket_name, jstring manager_socket_name,
+    jint release_delay) {
+  const char* s_name = env->GetStringUTFChars(store_socket_name, NULL);
+  const char* m_name = env->GetStringUTFChars(manager_socket_name, NULL);
+
+  plasma::PlasmaClient* client = new plasma::PlasmaClient();
+  ARROW_CHECK_OK(client->Connect(s_name, m_name, release_delay));
+
+  env->ReleaseStringUTFChars(store_socket_name, s_name);
+  env->ReleaseStringUTFChars(manager_socket_name, m_name);
+  return reinterpret_cast<int64_t>(client);
+}
+
+JNIEXPORT void JNICALL Java_org_apache_arrow_plasma_PlasmaClientJNI_disconnect(
+    JNIEnv* env, jclass cls, jlong conn) {
+  plasma::PlasmaClient* client = (plasma::PlasmaClient*)conn;
```

Comment:
Let's do a reinterpret_cast here, we shouldn't have old style casts

Reply:
ok